bundles/http/Citrix/Slack/2e8a194d-0c01-44c1-9a62-a5ee1d5ecbb3.js

```diff
@@ -0,0 +1,276 @@
+                "columns": [
+                    {
+                        "name": "id",
+                        "type": "STRING",
+                        "length": 255,
+                        "primaryKey": true
+                    },
+                    {
+                        "name": "user",
+                        "type": "STRING",
+                        "length": 255,
+                        "primaryKey": true
+                    },
+                    {
+                        "name": "name",
+                        "type": "STRING",
+                        "length": 255,
+
+                    },
+                    {
+                        "name": "ts",
+                        "type": "STRING",
+                        "length": 255,
+
+                    },
+                    {
+                        "name": "is_favorite",
+                        "type": "BOOLEAN"
+                    }
+                ]
+            }
+        ]
+    }
+});
+
+async function changeIDbyRealName(dataStore, client, isIncremental, latestSynchronizationTime=null) {
+    let respFavChannels = await client.fetch(ENDPOINT_FAVORITES_CHANNELS)
+    let parsedChannelMessages = [];
+    let parsedAllFavoriteChannels = [];
+    let parsedAllChannels = [];
+    let urlEndpoint = "";
+
+    if (respFavChannels.ok) {
+
+        let favoriteChannels = await respFavChannels.json()
+        let prettyFavoriteChannels = JSON.stringify(favoriteChannels.messages, null, '\t')
+        let parsedFavoriteChannels = JSON.parse(prettyFavoriteChannels)
+        let indexChannels = 0
+        let usedChannels = []
+        let nextChannel = false
+        let strLastSyncDateTime = ""
+
+        do {
+            urlEndpoint = ENDPOINT_FAVORITE_CHANNEL_MESSAGES
+
+            if (isIncremental) {
+                if (latestSynchronizationTime) {
+                    strLastSyncDateTime = latestSynchronizationTime
+                    let lastSyncDateTime = new Date(strLastSyncDateTime)
+                    urlEndpoint = urlEndpoint.replace(OLDEST_TIMESTAMP, `${lastSyncDateTime.getTime() / 1000}`.slice(0, 10))
+                } else {
+                    let sevenDaysBeforeSync = new Date();
+                    sevenDaysBeforeSync.setDate(sevenDaysBeforeSync.getDate() - 7);
+                    urlEndpoint = urlEndpoint.replace(OLDEST_TIMESTAMP, `${sevenDaysBeforeSync.getTime() / 1000}`.slice(0, 10))
+                }
+            }
+
+            usedChannels.forEach(channel => {
+                if (channel === parsedFavoriteChannels[indexChannels].text)
+                    nextChannel = true
+            });
+
+            if (nextChannel) {
+                nextChannel = false
+                indexChannels++
+                continue
+            }
+
+            urlEndpoint = urlEndpoint.replace(FAVORITE_CHANNEL, parsedFavoriteChannels[indexChannels].text)
+            let response = await client.fetch(urlEndpoint)
```

Comment:
Maybe a simpler and readable solution?
IMHO `replace` of constants is odd here

```js
let timestamp = OLDEST_TIMESTAMP
if (incrementalSync) {
    timestamp = ...
}

const channel = parsedFavoriteChannels[indexChannels].text

const response = await client.fetch(
    `/conversations.history?channel=${channel}&oldest=${timestamp}&limit=200`
)
```

and I guess that `FAVORITE_CHANNEL` is not needed here, right?

Reply:
You are right. We changed to the way you showed.